### ISSUE TITLE
Android Module - Is the mediaplayer being prepared on the UI thread?

### ISSUE THREAD

**Issue by davetao** [futurice/react-native-audio-toolkit#19]
I was wondering if the mediaplayer is being managed on the UI thread on the android side. This is because I am experiencing some UI lag when i call to prepare the audio player.

**Comment by ohtangza** [comment]
Talking about Android side, the *prepare()* function is called in the main UI thread, but it might not cause the delay as the native call of *prepare()* would have. One thing to consider is that Android natively supports *prepareAsync()* which seems to be better fit to react native port. 

I took a look at the relevant code, and the current implementation uses *prepare()* function (see: https://github.com/futurice/react-native-audio-toolkit/blob/4746870166fe4beb9fbf075d45fab952de4558d6/android/lib/src/main/java/com/futurice/rctaudiotoolkit/AudioRecorderModule.java)

```java
try {
            player.prepare();

            callback.invoke(null, getInfo(player));
        } catch (Exception e) {
            callback.invoke(errObj("prepare", e.toString()));
        }
```

I think we can improve it by using *prepareAsync()* function in this code block. If @FruitieX agrees, I might have time to work on this because it also affects my product's UX as well.

**Comment by FruitieX** [comment]
Absolutely, that's a bug if it's ran on the UI thread. If somebody has a fix I will be happy to accept it.

**Comment by dnish** [comment]
Have to agree, if I click on the song item, the animations stocks until the song is ready to play. I've added a short delay to the play function, it fixes the issue on my test device, I'll have to check it on other devices too:

    this.playerInstance = new NativePlayer(url);
        setTimeout(() => {
            this.playerInstance.play();
        },50);
